packages/shared/src/components/opportunity/PreferenceOptionsForm.tsx

```diff
@@ -11,37 +12,109 @@ import { Radio } from '../fields/Radio';
 import { Checkbox } from '../fields/Checkbox';
 import { Dropdown } from '../fields/Dropdown';
 import { TextField } from '../fields/TextField';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { getCandidatePreferencesOptions } from '../../features/opportunity/queries';
+import { RoleType } from '../../features/opportunity/types';
+import { LocationType } from '../../features/opportunity/protobuf/util';
+import {
+  EmploymentType,
+  SalaryPeriod,
+} from '../../features/opportunity/protobuf/opportunity';
+
+export const snapToHalf = (v: number): 0.0 | 0.5 | 1.0 => {
+  const x = Math.min(1, Math.max(0, v));
+
+  if (x < 0.25) {
+    return 0.0;
+  }
+  if (x < 0.75) {
+    return 0.5;
+  }
+
+  return 1.0;
+};
+
+const salaryDurationOptions = [
+  { label: 'Annually', value: SalaryPeriod.ANNUAL },
+  { label: 'Monthly', value: SalaryPeriod.MONTHLY },
+  { label: 'Hourly', value: SalaryPeriod.HOURLY },
+];
```

Comment:
Just a nice to have, we don't need it, but in case the role type gets stored as something other than `0.0`, `0.5` or `1.0`, we can make it snap to closet accepted value